posthog/temporal/batch_exports/monitoring.py

```diff
@@ -0,0 +1,220 @@
+            count=result.count,
+        )
+    activity.logger.info(f"Updated {total_rows_updated} BatchExportRuns")
+    return total_rows_updated
+
+
+@workflow.defn(name="batch-export-monitoring")
+class BatchExportMonitoringWorkflow(PostHogWorkflow):
+    """Workflow to monitor batch exports.
+
+    We have had some issues with batch exports in the past, where some events
+    have been missing. The purpose of this workflow is to monitor the status of
+    batch exports for a given customer by reconciling the number of exported
+    events with the number of events in ClickHouse for a given interval.
+    """
+
+    @staticmethod
+    def parse_inputs(inputs: list[str]) -> BatchExportMonitoringInputs:
+        """Parse inputs from the management command CLI."""
+        loaded = json.loads(inputs[0])
+        return BatchExportMonitoringInputs(**loaded)
+
+    @workflow.run
+    async def run(self, inputs: BatchExportMonitoringInputs):
+        """Workflow implementation to monitor batch exports for a given team."""
+        # TODO - check if this is the right way to do logging since there seems to be a few different ways
+        workflow.logger.info("Starting batch exports monitoring workflow for team %s", inputs.team_id)
+
+        batch_export_details = await workflow.execute_activity(
+            get_batch_export,
+            inputs.team_id,
+            start_to_close_timeout=dt.timedelta(minutes=1),
+            retry_policy=RetryPolicy(
+                maximum_attempts=3,
+                initial_interval=dt.timedelta(seconds=20),
+                non_retryable_error_types=["NoActiveBatchExportsFoundError"],
+            ),
+            heartbeat_timeout=dt.timedelta(minutes=1),
+        )
+
+        # time interval to check is not the previous hour but the hour before that
+        # (just to ensure all recent batch exports have run successfully)
+        now = dt.datetime.now(tz=dt.UTC)
+        interval_end = now.replace(minute=0, second=0, microsecond=0) - dt.timedelta(hours=1)
+        interval_start = interval_end - dt.timedelta(hours=1)
+        interval_end_str = interval_end.strftime("%Y-%m-%d %H:%M:%S")
+        interval_start_str = interval_start.strftime("%Y-%m-%d %H:%M:%S")
+
+        total_events = await workflow.execute_activity(
+            get_event_counts,
+            GetEventCountsInputs(
+                team_id=inputs.team_id,
+                interval=batch_export_details.interval,
+                overall_interval_start=interval_start_str,
+                overall_interval_end=interval_end_str,
+                exclude_events=batch_export_details.exclude_events,
+                include_events=batch_export_details.include_events,
+            ),
+            start_to_close_timeout=dt.timedelta(hours=1),
+            retry_policy=RetryPolicy(maximum_attempts=3, initial_interval=dt.timedelta(seconds=20)),
```

Comment:
We probably want to exponentially backoff on retrying. As this will be hitting ClickHouse, the most common error will be a `TOO_MANY_SIMULTANEOUS_QUERIES` error from ClickHouse. This workflow should make sure it's not putting too much load in ClickHouse by being very generous with the retry interval when retrying.

Reply:
Sounds good, although I think there is a default `backoff_coefficient` of 2 so currently this will retry after 20s, then again after 40s, then finally after 80s. Do you think it should be increased more?

Reply:
Oh yeah, you are correct.